RIOT-OS/RIOT
# cpu/rpx0xx: port RIOT to the Raspberry Pi RP2040 MCU

### Contribution description

This provides:

- Support for the RP2040 MCU
     - Only `periph_gpio`, `periph_gpio_int`, and `periph_uart` are provided in this PR. Other than that no timers, no SPI, no nothing is provided.
     - Work on additional drivers can more easily be parallelized once basic support is upstream, which this PR tries to do.
- Support for the Raspberry Pi Pico board (`rpi-pico`), the reference board for the RP2040 MCU, is provided

### Acknowledgements

This code was developed as part of a software project in a combined effort of students and mentors. Thanks to @nickw96, @MaestroOnICe, and @Franz2000.

Most of the low level code and especially the boot sequence and clock configuration was developed by @fabian18.

### Testing procedure

The following test applications will be useful:

- `examples/default`
    - stdio via UART should work
    - Via SAUL the on-board LED should be controllable
- `tests/periph_gpio`
    - The GPIO pins should work as input (floating, with pull up, with pull down) and output and IRQs should be supported on rising, falling, and both edges

It would be useful to test all of the flashing options:

- Using OpenOCD and `pico-debug` (currently the default option)
     - Beware: First "flash" the gimmechace variant from here https://github.com/majbthrd/pico-debug/releases via the UF2 bootloader into RAM. (Needs to be re-"flashed" after each cold boot, or RAM corruption.)
- Using J-Link
- Using the UF2 bootloader

### Issues/PRs references

~None~ closes #15822
